src/Compilers/Core/Portable/Binding/AbstractLookupSymbolsInfo.cs

```diff
@@ -181,12 +183,12 @@ private void AddArity(int arity)
                 hashSet.Add(arity);
             }
 
-            public void GetUniqueSymbolOrArities(out IArityEnumerable arities, out TSymbol uniqueSymbol)
+            public void GetUniqueSymbolOrArities(out IArityEnumerable? arities, out TSymbol? uniqueSymbol)
             {
                 if (this.HasUniqueSymbol)
                 {
                     arities = null;
-                    uniqueSymbol = (TSymbol)_uniqueSymbolOrArities;
+                    uniqueSymbol = (TSymbol)_uniqueSymbolOrArities!;
```

Comment:
Consider adding an assert instead of using a suppression

Reply:
There should not be any new assertions added in my current NRT pull requests (they were too difficult to validate for PRs at this scale). I'm using one of the following two approaches in each relevant case:

1. If the state can be "trivially determined" from context (looking at most one symbol deep), there might be a suppression.
2. In other cases, I have a `#nullable disable` with a comment regarding the item in question.

Reply:
Suppressions can have lasting consequences in that they'll cause future warnings to be suppressed, rather than correctly flagged. I only saw 2 instances in this PR, so it should be easy to add the appropriate assertions.

Reply:
I'll review this pull request to find suppressions that can't be directly verified from the current method (without looking to the definitions of other members), and convert those cases to the second form.

Reply:
The suppression decisions have now been deferred for future work.